src/geometries/multigeom.jl

```diff
@@ -20,7 +20,12 @@ struct Multi{M<:Manifold,C<:CRS,G<:Geometry{M,C}} <: Geometry{M,C}
 end
 
 # constructor with iterator of geometries
-Multi(geoms) = Multi(collect(geoms))
+
+Multi(geoms) = begin
+  all(gT <: Geometry{M,C} where {M<:Manifold,C<:CRS} for gT in typeof.(geoms)) ||
+  throw("All elements used must be subtypes of the same Geometry{<:Manifold, <:CRS}")
+  Multi(collect(geoms))
+end
```

Comment:
Please remove this change, it is unrelated to the issue, and we want to support heterogeneous geometry lists.

Reply:
`gT <: Geometry{M, C}` matches everything that is a subtype of geometry. It's not enforcing the same type for all.
Could have been worded better. This is off topic though. removed as requested, I'll make a separate PR if desired.